### ISSUE TITLE
Public API

### ISSUE THREAD

**Issue by jesperlindstrom** [cthit/hubbIT-old#28]
A public developer API would allow us to write fun services such as notifications for ranking changes and warnings ("X is about to pass you in the toplist. Go to Hubben immediately!").

Read-only access would be sufficient.

**Comment by SSundstrom** [comment]
+1

**Comment by SamuelHakansson** [comment]
+1

**Comment by lindskogen** [comment]
Rails provides us with a REST API automatically, so most of this is already done. I would think that a lot of optimizations is required before we can open up this publicly.

The API is already available right now for signed in users, just add `'.json'` to the url!

**Comment by jesperlindstrom** [comment]
I'm signed in, but I'm getting internal server error at https://hubbit.chalmers.it/stats.json

``` json
{
"status": "500",
"error": "Internal Server Error"
}
```

**Comment by lindskogen** [comment]
@jesperlindstrom Now you can :)

**Comment by Edholm** [comment]
I'm closing this since we won't make the API any more public than it already is...

Feel free to reopen if you disagree

**Comment by Dr-Horv** [comment]
We should add read-only-api-tokens though so users can request them and be granted if deemed ok. So users doesn't have to rely on their cth.it cookie authentication.

**Comment by lindskogen** [comment]
Closing this since this is not really what the issue is about
